src/native/corehost/error_codes.h

```diff
@@ -51,6 +51,8 @@ enum StatusCode
     HostPropertyNotFound                = 0x800080a4,
     CoreHostIncompatibleConfig          = 0x800080a5,
     HostApiUnsupportedScenario          = 0x800080a6,
+    CorehostMustRunAsArchX64            = 0x800080a7,
+    CorehostMustRunAsArchArm64          = 0x800080a8,
```

Comment:
The error codes are also documented here: https://github.com/dotnet/runtime/blob/main/docs/design/features/host-error-codes.md

I know that these two should not ever be reported externally, but it's a good idea to document them anyway.

Reply:
Actually they are visible externally if using hosting APIs (since in that case we can't switch the arch of the process).